### DISCUSSION TITLE
SSL support for the docker image

### DISCUSSION THREAD

**Discussion by yarathenomad** [SillyTavern/SillyTavern#3167] [General]
I'm wondering whether SSL/HTTPS is supported in the docker and if not how it could be implemented. Right now all the documentation for HTTPS [that I can find](https://docs.sillytavern.app/usage/remoteconnections/#https) involves starting the node server with the `--ssl` flag. On the other hand the docker image starts the node server from within `docker/docker-endpoint.sh` with:
```
exec node server.js --listen
```
I think the most straightforward way to add SSL to the docker is to add the `--ssl` flag to that script dependent on an environment variable that, as far as I can see, can be set from the Dockerfile itself.

Am I missing anything? Is this something you want to see happen? If it works locally I can open a PR for it.

**Comment by Cohee1207** [comment]
Since you're already using Docker, it would ideally be for you to be using an nginx server container in front of the SillyTavern container that would be handling HTTPS/SSL, including redirection from HTTP and certificate management. Something I've found with a quick search: https://github.com/nginx-proxy/nginx-proxy

**Comment by yarathenomad** [comment]
I've thought of that and it works as a generic way to get HTTPS when you don't want to think about adding it to an existing application. But after playing around, ST already has pretty decent HTTPS support. I've [opened a PR](https://github.com/SillyTavern/SillyTavern/pull/3169) that allows using the docker image with the same args as a baremetal installation.

**Reply by Cohee1207** [reply]
HOW TO for future reference:
1. Put a private key and a certificate in your config volume.
2. Modify the docker-compose.yml as follows:
```diff
diff --git a/docker/docker-compose.yml b/docker/docker-compose.yml
index 1431c9d8b..ea3df066a 100644
--- a/docker/docker-compose.yml
+++ b/docker/docker-compose.yml
@@ -5,6 +5,7 @@ services:
     container_name: sillytavern
     hostname: sillytavern
     image: ghcr.io/sillytavern/sillytavern:latest
+    command: ["--ssl", "true", "--certPath", "config/localhost.crt", "--keyPath", "config/localhost.key"]
     ports:
       - "8000:8000"
     volumes:
```
